python/cog/director/director.py

```diff
@@ -309,6 +315,32 @@ def _handle_message(self, message_id, message_json) -> None:
             log.info("prediction succeeded")
             self._record_success()
 
+    def _confirm_model_health(self):
+        self.healthchecker.request_status()
+        mark = time.perf_counter()
+
+        while time.perf_counter() - mark < HEALTHCHECK_WAIT:
+            try:
+                event = self.events.get(timeout=POLL_INTERVAL)
+            except queue.Empty:
+                continue
+
+            if not isinstance(event, HealthcheckStatus):
+                log.warn(
```

Comment:
The only other event we can get here is a webhook, which would indicate something is *very* wrong. I think this is probably fine as is, but maybe something to keep an eye on and make more defensive (ie abort) if we see it being an issue in prod.

Reply:
Yup, I went back and forth on this one. I guess let's see what happens in prod. If we are regularly getting webhook events here we probably need to dig into why in addition to possibly crashing here instead.